### PR TITLE
Feature - Adoption register screen

### LINKED ISSUE #76: Create adoption register screen
## Overview
The screen for users register an adoption.

#30

### PR DESCRIPTION
**Pull request by petruspierre**
closes #76 

Description:
Created a custom hook to handle the adoption register using context API and react hook form

DEMO:

https://user-images.githubusercontent.com/39802870/139121584-4ed0f9aa-e2d0-4e38-9894-ab59b6a35504.mp4

### CODE REVIEW

File: packages/expo/src/modules/adoption/AdoptionForm.tsx [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+`;
+
+const AdoptionForm: React.FC = () => {
+  const {RegisteredInput, submitData, errors} = useAdoptionRegister();
+  const {goBack} = useNavigation();
+  const {t} = useI18n(['register_adoption']);
+
+  const formFields = getForm(t);
+
+  const getInputElement = (data: InputData) => {
```
Comment (belgamo) [inline packages/expo/src/modules/adoption/AdoptionForm.tsx]: I think you can place this outside of the component.

File: packages/expo/src/modules/adoption/AdoptionForm.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  const renderError = (name: string) => {
+    const error = errors[name];
+
+    if (error) {
+      return (
+        <Typography css={[`margin-bottom: ${px2ddp(4)}px;`]} variant="caption-1">
+          {t('required_field')}
+        </Typography>
+      );
+    } else {
```
Comment (belgamo) [inline packages/expo/src/modules/adoption/AdoptionForm.tsx]: How about adding an earlier return? Like:

```js
if(!error) return;

const error = errors[name];

return (
   <Typography css={[`margin-bottom: ${px2ddp(4)}px;`]} variant="caption-1">
      {t('required_field')}
   </Typography>
);
```

File: packages/expo/src/modules/adoption/Controller.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+const AdoptionContext = createContext<AdoptionContextProps>({} as AdoptionContextProps);
+
+const AdoptionProvider: React.FC<Partial<AdoptionContextProps>> = props => {
+  const {children} = props;
+
+  const [status, setStatus] = useState(ImagePicker.PermissionStatus.UNDETERMINED);
+  const [step, dispatchStep] = useReducer(stepsReducer, 0);
+  const [images, dispatchImages] = useReducer(imagesReducer, []);
+  const [data, setData] = useState<Partial<Fields>>({});
```
Comment (belgamo) [inline packages/expo/src/modules/adoption/Controller.tsx:41]: In order to make it easier to understand, I suggest transforming the data state into a `useReducer`. You have two small reducers for step and another one for images. Also, I think you don't need a data state to store form info. So I think would be great to unify all reducers and the data state. What do you think?

File: packages/expo/src/modules/adoption/__tests__/AdoptionForm.test.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import {fireEvent} from '@testing-library/react-native';
+
+import {AdoptionProvider, AdoptionContextProps} from '../Controller';
+import AdoptionForm from '../AdoptionForm';
+import {Mount} from '../../../../tests/helpers';
```
Comment (belgamo) [inline packages/expo/src/modules/adoption/__tests__/AdoptionForm.test.tsx]: Use absolute import here, please.

File: packages/expo/src/modules/adoption/reducers.ts [hunk 5]
```diff
@@ -0,0 +1,64 @@
+import * as ImagePicker from 'expo-image-picker';
```
Comment (belgamo) [inline packages/expo/src/modules/adoption/reducers.ts:1]: Missing unit tests for reducers. See my comment above about unifying all reducers in just a single one.

File: packages/ui/assets/images/PhotoContainerImage.tsx [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import PhotoContainerImageBinaries from './binaries/PhotoContainerImage.jpg';
+
+const Image = styled.Image`
+  width: 100%;
+  height: ${px2ddp(120)}px;
+`;
+
+const PhotoContainerImage: React.FC = () => {
+  // @ts-ignore
```
Comment (belgamo) [inline packages/ui/assets/images/PhotoContainerImage.tsx]: Why do we need this here?

File: packages/ui/core/Button/__tests__/ButtonGroup.test.tsx [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+const mountFactory = propOverrides => {
+  const buttonProps = {...defaultProps, ...propOverrides};
+
+  const element = Mount(<ButtonGroup {...buttonProps} />);
+
+  return element;
+};
+
+describe('Header component', () => {
```
Comment (belgamo) [inline packages/ui/core/Button/__tests__/ButtonGroup.test.tsx]: I think you mean "ButtonGroup component"?

**Review by belgamo — CHANGES_REQUESTED** [review]
Thank you for helping us a lot with excellent PR!

File: packages/ui/assets/images/PhotoContainerImage.tsx (hunk 6 above)
Comment (petruspierre) [reply packages/ui/assets/images/PhotoContainerImage.tsx]: forgot to remove it xD, thanks

**Pull request closed without merging**